### PR TITLE
Add FreeBSD support

### PR DESCRIPTION
**Pull request by rozhuk-im**
Add FreeBSD support and probably fix other OSes to not show network interfaces without BROADCAST flag.

### CODE REVIEW

**Comment by rozhuk-im** [comment]
Have no idea why Linux and MSVC fail, looks like some temporary errors in test system.

**Comment by rozhuk-im** [comment]
After I disable ```if ((ifap_iter->ifa_flags & IFF_BROADCAST) == 0)``` test is passed, but IMHO this is wrong, and interfaces without broadcast flag must not be used.

File: src/arvnetwork.c [hunk 1]
```diff
@@ -354,6 +354,10 @@ arv_enumerate_network_interfaces (void)
 		return NULL;
 
 	for (ifap_iter = ifap; ifap_iter != NULL; ifap_iter = ifap_iter->ifa_next) {
+//#ifdef IFF_BROADCAST
+//		if ((ifap_iter->ifa_flags & IFF_BROADCAST) == 0)
+//			continue; /* Skip interfaces without broadcast. */
+//#endif
```
Comment (EmmanuelP) [inline src/arvnetwork.c:360]: Please remove this part of commented code.

**Review by EmmanuelP — CHANGES_REQUESTED** [review]
Thanks for the patch. Once the commented code is removed, it will be ready to merge.

**Comment by EmmanuelP** [comment]
> After I disable `if ((ifap_iter->ifa_flags & IFF_BROADCAST) == 0)` test is passed, but IMHO this is wrong, and interfaces without broadcast flag must not be used.

That is not really related to the main goal of this pull request, so the pull request can be merged without this part. But feel free to investigate why this test makes the fakegv test fail on linux and macOS in another pull request.

File: src/arvnetworkprivate.h [hunk 2]
```diff
@@ -38,16 +39,22 @@
...
 #endif
 
-#if !defined(__APPLE__) && !defined(G_OS_WIN32)
+#ifdef __linux__
 #include <linux/ip.h>
 #endif
 
+#if defined(__APPLE__) || defined(BSD)
+#include <netinet/in.h>
+#include <netinet/ip.h>
```
Comment (EmmanuelP) [inline src/arvnetworkprivate.h:48]: Redundant include.

**Review by EmmanuelP — CHANGES_REQUESTED** [review]

**Comment by EmmanuelP** [comment]
Thanks a lot.

**Pull request merged**